nixos/modules/services/matrix/maubot.nix

```diff
@@ -0,0 +1,445 @@
+{ lib
+, config
+, pkgs
+, ...
+}:
+
+let
+  cfg = config.services.maubot;
+
+  wrapper1 =
+    if cfg.plugins == [ ]
+    then cfg.package
+    else cfg.package.withPlugins (_: cfg.plugins);
+
+  wrapper2 =
+    if cfg.pythonPackages == [ ]
+    then wrapper1
+    else wrapper1.withPythonPackages (_: cfg.pythonPackages);
+
+  settings = cfg.settings // {
+    plugin_directories = cfg.settings.plugin_directories // {
+      trash =
+        if cfg.settings.plugin_directories.trash == null
+        then "delete"
+        else cfg.settings.plugin_directories.trash;
+    };
+    server = cfg.settings.server // {
+      unshared_secret = "generate";
+    };
+  };
+
+  finalPackage = wrapper2.withBaseConfig settings;
+
+  isPostgresql = db: builtins.isString db && lib.hasPrefix "postgresql://" db;
+  isLocalPostgresDB = db: isPostgresql db && builtins.any (x: lib.hasInfix x db) [
+    "@127.0.0.1/"
+    "@::1/"
+    "@localhost/"
+  ];
+  parseLocalPostgresDB = db:
+    let
+      noSchema = lib.removePrefix "postgresql://" db;
+      username = builtins.head (lib.splitString "@" noSchema);
+      database = lib.last (lib.splitString "/" noSchema);
+    in
+    if lib.hasInfix ":" username then null else {
```

Comment:
Why not split at `:` here to separate out the username? Especially that the database options use `postgresql://username:password@hostname/dbname` as example.

Reply:
in fact I don't want people putting their passwords in the Nix store, so I'll just add an assertion